### PR TITLE
MODRR-11 Create POST API for RRA

### PR DESCRIPTION
**Pull request by Vignesh-kalyanasundaram**
## Purpose
The purpose of the PR is to implement [MODRR-11](https://folio-org.atlassian.net/browse/MODRR-11)

## Approach
We have created a new Post endpoint and implement the validation for readingroom and servicepoints. Exception handling is implemented. 

#### TODOS and Open Questions
- [ ] Use GitHub checklists. When solved, check the box and explain the answer.

## Learning
_Describe the research stage. Add links to blog posts, patterns, libraries or addons used to solve this problem._

### CODE REVIEW

File: src/main/java/org/folio/readingroom/client/feign/InventoryServicePointClient.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import org.folio.spring.config.FeignClientConfiguration;
+import org.folio.spring.model.ResultList;
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+
+@FeignClient(name = "service-points", configuration = FeignClientConfiguration.class)
+public interface InventoryServicePointClient {
+
+  @GetMapping("?query=id==({servicePointIds})")
```
[the remark this answers is not in the document]
Comment (Vignesh-kalyanasundaram) [reply src/main/java/org/folio/readingroom/client/feign/InventoryServicePointClient.java]: Done

File: src/main/java/org/folio/readingroom/controller/ReadingRoomController.java [hunk 2]
```diff
@@ -1,9 +1,27 @@
...
+@AllArgsConstructor
+@Log4j2
 public class ReadingRoomController implements ReadingRoomApi {
 
+  private final ReadingRoomService readingRoomService;
+
+  @Override
+  public ResponseEntity<ReadingRoom> createReadingRoom(ReadingRoom readingRoom) {
+    log.info("createReadingRoom:: creating reading room with name {}, id {}",
+      readingRoom.getId(), readingRoom.getName());
```
[the remark this answers is not in the document]
Comment (Vignesh-kalyanasundaram) [reply src/main/java/org/folio/readingroom/controller/ReadingRoomController.java:22]: Done

File: src/main/java/org/folio/readingroom/service/impl/ReadingRoomServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+  private void validateReadingRoom(UUID readingRoomId) {
+    log.debug("validateReadingRoom:: validating readingRoom with id {}", readingRoomId);
+    readingRoomRepository.findById(readingRoomId)
+      .ifPresent(entity -> {
+        throw new ResourceAlreadyExistException("Reading room with id " + readingRoomId + " already exists");
+      });
+  }
+
+  private void validateServicePoints(Set<ServicePoint> servicePointDtoList) {
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/readingroom/service/impl/ReadingRoomServiceImpl.java:49]: servicePointDtoList the variable name needs to changed as its not a list now..

**Review by gurleenkaurbp — APPROVED** [review]

**Review by SinghAdes — APPROVED** [review]

File: src/main/java/org/folio/readingroom/service/impl/ReadingRoomServiceImpl.java (hunk 3 above)
Comment (Vignesh-kalyanasundaram) [reply src/main/java/org/folio/readingroom/service/impl/ReadingRoomServiceImpl.java:49]: I will rename this in next PR

**Pull request merged**